pkg/audit/manager.go

```diff
@@ -242,11 +242,11 @@ func (am *AuditManager) writeAuditResults(ctx context.Context, resourceList *met
 		for _, item := range instanceList.Items {
 			updateConstraints[item.GetSelfLink()] = item
 
-			// report total violations and constraint counts
-			am.reporter.ReportTotalViolations(item.GetKind(), item.GetName(), totalViolations[item.GetSelfLink()])
-			am.reporter.ReportConstraints(item.GetKind(), int64(len(instanceList.Items)))
+			// report constraint counts and total violations
+			enforcementAction := updateLists[item.GetSelfLink()][0].enforcementAction
```

Comment:
EnforcementAction is an open string, IIUC? If so then the cardinality of this tag is unbounded and the value is potentially private data, which are both problems. From talking to Max, my understanding is that gatekeeper officially supports three values: "deny", "audit", and "dryrun". I would normalize any other value to "unrecognized" so that you keep the cardinality bounded and don't risk a privacy leak.

Reply:
enforcementAction can be either `deny` (default), `dryrun`, but user can disable validation to specify any string. Added validation to classify as `deny`, `dryrun` or `unrecognized`.